api/src/main/java/org/apache/iceberg/Table.java

```diff
@@ -44,6 +44,28 @@
    */
   TableScan newScan();
 
+
+  /**
+   * Create a table scan which can read append data from {@param fromSnapshotId}
+   * exclusive and up to {@param toSnapshotId} inclusive
+   * @param fromSnapshotId - the last snapshot id read by the user, exclusive
+   * @param toSnapshotId - read append data up to this snapshot id
+   * @return a table scan which can read append data from {@param fromSnapshotId}
+   * exclusive and up to {@param toSnapshotId} inclusive
+   */
+  TableScan appendsBetween(long fromSnapshotId, long toSnapshotId);
```

Comment:
@rdsr, overall I like how much cleaner this has become! Thanks for working on making it simpler.

Now that this has evolved quite a bit, I think it makes sense for `IncrementalDataScan` to be created through `newScan()` just like any other data scan. This would mean fewer new methods on `Table` and that's always a good thing. `IncrementalDataScan` already extends `DataTableScan` and has to handle conflicts between `appendsBetween` and `asOfTime` and `useSnapshot`, so it isn't too much complication. I think we should add these methods to the `TableScan` interface.

We can still have a separate `IncrementalDataScan` class if that's cleaner. The `TableScan` implementation classes are all hidden anyway.

Reply:
Thanks, I'll try this out.

Reply:
Fixed